lib/package.js

```diff
@@ -164,6 +164,13 @@
           skipExtension = true;
       });
 
+    // exact global meta or meta with any content after the last wildcard skips extension
+    if (!skipExtension && loader.meta)
+      getMetaMatches(loader.meta, pkgName, subPath, function(metaPattern, matchMeta, matchDepth) {
```

Comment:
`subPath` can be replaced with `pkgName + '/' + basePath + subpath`

Also the `pkgName` argument to `getMetaMatches` can be entirely removed from the signature.